.gitmodules

```diff
@@ -26,3 +26,9 @@
 [submodule "scipy/_lib/cobyqa"]
 	path = scipy/_lib/cobyqa
 	url = https://github.com/cobyqa/cobyqa.git
+[submodule "scipy/_lib/prima"]
```

Comment:
small comment from the sidelines - https://github.com/scipy/scipy/issues/18118#issuecomment-1824123053 suggested that we include the submodule

> in a top-level `subprojects/` directory

Reply:
At one point we were going to switch PRIMA over to Meson from CMake and take this approach, but as I sat down to do the switch I found I didn't much care for Meson and started to take a look at how other projects have integrated with SciPy. It seems there is no subprojects directory anywhere in SciPy, and almost all submodules are in _lib. It seemed like the easiest thing to do both for this current project and for the long term would be to just follow the existing pattern.

Reply:
Yes, it's fine to keep it in `_lib` in this PR. PRIMA is quite small and easy to understand build-wise, and I don't think there will be a desire from distros to unvendor it. So I'd rather not treat it as a subproject here (the likes of Boost.Math and HiGHS are in a different category, and we indeed do want to move those to subprojects).